### PR TITLE
Add an overload to the recordException method to support additional attributes

### PR DESCRIPTION
**Pull request by dengliming**
closes [https://github.com/open-telemetry/opentelemetry-java/issues/1597](https://github.com/open-telemetry/opentelemetry-java/issues/1597)

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]
Thanks!

**Review by iNikem — APPROVED** [review]

File: sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 1]
```diff
@@ -428,11 +428,25 @@ public void setStatus(Status status) {
...
+  }
+
+  @Override
+  public void recordException(Throwable exception, Attributes additionalAttributes) {
     if (exception == null) {
       return;
     }
     long timestamp = clock.now();
-    Attributes.Builder attributes = Attributes.newBuilder();
+    final Attributes.Builder attributes = Attributes.newBuilder();
```
Comment (jkwatson) [inline sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:440]: why not just assign this to `additionalAttributes.newBuilder()` if the additional attributes are provided? I think the code will read more simply and be slightly more efficient.

**Comment by jkwatson** [comment]
one small simplification, and this will be good to go. Thanks for picking this up!

File: sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java (hunk 1 above)
Comment (dengliming) [reply sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java]: 👍 Thanks. forgot this. : )

**Review by carlosalberto — APPROVED** [review]

File: sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 2]
```diff
@@ -428,11 +428,19 @@ public void setStatus(Status status) {
...
+  @Override
+  public void recordException(Throwable exception, Attributes additionalAttributes) {
     if (exception == null) {
       return;
     }
     long timestamp = clock.now();
-    Attributes.Builder attributes = Attributes.newBuilder();
+    Attributes.Builder attributes =
+        additionalAttributes != null
+            ? Attributes.newBuilder(additionalAttributes)
```
Comment (jkwatson) [inline sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:442]: even simpler is just `? addiontalAttributes.toBuilder()` yes?

Comment (dengliming) [reply sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java]: yes. fixed at lastest commit. thanks.

**Pull request merged**